src/validation.ts

```diff
@@ -186,38 +199,66 @@ export function is<T>(value: Value, validator: Validator<T>): value is T {
   return validator.validate(value).ok
 }
 
+function withSameMeta<A, B>(a: A, b: B): B {
+  const source = a as any
+  const target = b as any
+
+  Object.keys(source).forEach(k => {
```

Comment:
Instead of having this naming convention, how about using a namespace, i.e validator.meta where all the meta keys are found? then it's just a matter of copying that namespace in a single operation.